### PR TITLE
Feature/294 implement update highlight

### PR DESCRIPTION
**Pull request by matir8**
## Description
- Added update highlight functionality while reusing the existing highlight form
- Added pipe for formatting dates
- Renamed the `modal-highlight` component to `modal-highlight-form`

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

1. Log in as a user who has the permissions to edit the highlights
2. Go to the project overview page
3. Click the "Edit Highlight" button
4. Select the highlight for this project that you want to edit
5. Edit the information
6. Click the "Save" button
7. If the information is valid you should see a message that the highlight has been updated successfully

## Link to issue

Closes: #294

### CODE REVIEW

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
I was testing this functionality and have the following feedback:
- When showing the lists of highlights, please also show part of the description (just cut it off with '...' and perhaps show it fully on hover in a tooltip?
- When clicking on a highlight to edit it (one that has a start/end date), the date is not shown in the input fields. It will say dd/mm/yyyy instead of the date that was previously there.
- Try to be uniform in the styling, either make the 'delete highlight' list view the same as this update one or the other way around. (one has checkboxes, the other does not and uses on-click).
- While you're at it, can you also update 'delete highlight' listview with a description from the highlight? Thank you!

**Comment by matir8** [comment]
@Brend-Smits Since [this issue](https://github.com/DigitalExcellence/dex-backend/issues/205) is implemented now we don't have the start and the end dates available on the highlights endpoint. Why was it necessary to remove them?

**Comment by matir8** [comment]
@Brend-Smits Also since we are going to redesign the Project Detail Page #323, maybe we need to add a concrete defined design for these modals as well before we implement and ship it. What do you think?

**Comment by Brend-Smits** [comment]
> @Brend-Smits Since [this issue](https://github.com/DigitalExcellence/dex-backend/issues/205) is implemented now we don't have the start and the end dates available on the highlights endpoint therefore we can't display them on the edit highlight component. Why was it necessary to remove them?

That's definitely an unintended side effect... I will fix it in the backend of this release. Thanks for noticing that.

**Comment by Brend-Smits** [comment]
> @Brend-Smits Also since we are going to redesign the Project Detail Page #323, maybe we need to add a concrete defined design for these modals as well before we implement and ship it. What do you think?

I agree, I will add it to the issue.

File: src/app/modules/project/highlights-modal/highlights-modal.component.ts
Comment (Ruby77) [inline src/app/modules/project/highlights-modal/highlights-modal.component.ts:51]: The code looks good! One thing I noted, I think (almost) every method in the typescript files has some sort of documentation. I suggest doing the same 😄 . For the rest: Good job

**Review by Ruby77 — CHANGES_REQUESTED** [review]
One small note

**Review by Ruby77 — COMMENTED** [review]
Code looks good and the implementation of the design also looks good in my eyes. Good job! 

I have 2 really small notes (it isn't really a problem I think... 😉 )

In my opinion, there is some ambiguity in your pull request description. `Click the "Edit Highlight" button`, In the current design I don't see a highlight button, I think this has been replaced by clicking on the highlight you want to edit.

Another thing I noticed, is when I click 2 times on the save button rapidly, the highlight is added twice. I don't know if this is really a big problem, but I want to validate that this is a known occurrence. 
image

**Comment by matir8** [comment]
@Ruby77 
1) I forgot to edit the PR description after we made changes to the design, sorry about that.
2) That's a good catch, we have to disable the button once clicked. I noticed that happens on other buttons as well. For example, when we add a project manually. I will create an issue for this.

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
I'm blocking this until #379 is merged.

**Review by Ruby77 — APPROVED** [review]
Great work, all my feedback is resolved😄

**Comment by Brend-Smits** [comment]
Project Detail was merged, can you fix the conflicts @matir8 ?

**Comment by liekevdvoort** [comment]
The highlight popup should have it's own overlay.

File: src/app/modules/project/highlights-modal/highlights-modal.component.scss
Comment (Brend-Smits) [inline src/app/modules/project/highlights-modal/highlights-modal.component.scss:58]:
```suggestion
    color: $light-mode-grey-secondary;
    
    &:hover {
      color: $accent-color-red-secondary;
    }
```

Comment (Brend-Smits) [reply src/app/modules/project/highlights-modal/highlights-modal.component.scss:58]: Adds a small hover over color to let the user know that it's a button they can click.

**Review by Brend-Smits — APPROVED** [review]
Looks great, nice work :) 
Could you please explain what a debounce click is? I noticed you reference a new package for it. 

Also, @liekevdvoort , could you take a look at the SCSS/Styling?

File: src/app/modules/project/highlights-modal/highlights-modal.component.scss
Comment (matir8) [reply src/app/modules/project/highlights-modal/highlights-modal.component.scss:58]: > Another thing I noticed, is when I click 2 times on the save button rapidly, the highlight is added twice. I don't know if this is really a big problem, but I want to validate that this is a known occurrence.

@Brend-Smits debounce click fixes this issue that @Ruby77 mentioned in the discussion above

Comment (Brend-Smits) [reply src/app/modules/project/highlights-modal/highlights-modal.component.scss:58]: Oh I see, that's pretty cool. Thanks.

**Review by Brend-Smits — APPROVED** [review]

**Comment by liekevdvoort** [comment]
Looks good to me

**Review by liekevdvoort — APPROVED** [review]

**Pull request merged**
